extensions/answer_summarizers/models_test.py

```diff
@@ -34,455 +46,320 @@ def _create_sample_answer(
             'classification_categorization': classify_category,
         }
 
-    def _create_sample_answers(
-            self, repeated_answer, times_spent_in_card, session_ids,
-            repeat_count):
-        """This is similar to _create_sample_answer, except it repeats a single
-        answer N times. It reuses times_spent_in_card and session_ids
-        cyclically as it constructs the list of sample answers.
+    def _create_answer_dicts_list(
+            self, answers, times_spent_in_card, session_ids,
+            classify_categories=(exp_domain.EXPLICIT_CLASSIFICATION,),
+            num=None):
+        """This is similar to _create_answer_dict, except it provides a list
+        of N answers. It reuses answers, times_spent_in_card, and session_ids
+        cyclically to construct the list. When num isn't provided, the returned
+        list will be len(answers).
         """
-        return [self._create_sample_answer(
-            repeated_answer,
-            times_spent_in_card[i % len(times_spent_in_card)],
-            session_ids[i % len(session_ids)]) for i in range(repeat_count)]
-
-    def test_answer_frequencies_calculation(self):
-        """For multiple choice interactions, test if the most common answers
-        are calculated correctly for interaction answer views.
-        """
-        # Some answers.
-        dummy_submitted_answer_list = [
-            self._create_sample_answer('First choice', 4., 'sid1'),
-            self._create_sample_answer('Second choice', 5., 'sid1'),
-            self._create_sample_answer('Fourth choice', 2.5, 'sid1'),
-            self._create_sample_answer('First choice', 10., 'sid2'),
-            self._create_sample_answer('First choice', 3., 'sid2'),
-            self._create_sample_answer('First choice', 1., 'sid2'),
-            self._create_sample_answer('Second choice', 20., 'sid2'),
-            self._create_sample_answer('First choice', 20., 'sid3')
+        if num is None:
+            num = len(answers)
+        inf_args = itertools.izip(
+            itertools.cycle(answers), itertools.cycle(times_spent_in_card),
+            itertools.cycle(session_ids), itertools.cycle(classify_categories))
+        return [
+            self._create_answer_dict(*a)
+            for a in itertools.islice(inf_args, num)
```

Comment:
Try to use more meaningful dummy variable names, e.g. answer_dict_args

Reply:
Done.